Makefile

```diff
@@ -10,22 +10,25 @@ MANPREFIX = $(PREFIX)/share/man
 AUTORELOAD = inotify
 
 # enable features requiring giflib (-lgif)
-HAVE_GIFLIB = 1
-
+HAVE_LIBGIF ?= 0
 # enable features requiring libexif (-lexif)
-HAVE_LIBEXIF = 1
-
-cflags = -std=c99 -Wall -pedantic $(CFLAGS)
-cppflags = -I. $(CPPFLAGS) -D_XOPEN_SOURCE=700 \
-  -DHAVE_GIFLIB=$(HAVE_GIFLIB) -DHAVE_LIBEXIF=$(HAVE_LIBEXIF) \
+HAVE_LIBEXIF ?= 0
+
+ifndef OPTIONAL_LIBS
+	ifeq ($(HAVE_LIBEXIF), 1)
+		OPTIONAL_LIBS += -lexif
+	endif
+	ifeq ($(HAVE_LIBGIF), 1)
+		OPTIONAL_LIBS += -lgif
+	endif
+endif
+
+CFLAGS ?= -std=c99 -Wall -pedantic
+CFLAGS += -D_XOPEN_SOURCE=700 \
+  -DHAVE_LIBGIF=$(HAVE_LIBGIF) -DHAVE_LIBEXIF=$(HAVE_LIBEXIF) \
   -I/usr/include/freetype2 -I$(PREFIX)/include/freetype2
 
-lib_exif_0 =
-lib_exif_1 = -lexif
-lib_gif_0 =
-lib_gif_1 = -lgif
-ldlibs = $(LDLIBS) -lImlib2 -lX11 -lXft -lfontconfig \
-  $(lib_exif_$(HAVE_LIBEXIF)) $(lib_gif_$(HAVE_GIFLIB))
+LDFLAGS = -lImlib2 -lX11 -lXft -lfontconfig $(OPTIONAL_LIBS)
```

Comment:
We should use `LDLIBS` here.

https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html
>LDFLAGS
> Extra flags to give to compilers when they are supposed to invoke the linker, ‘ld’, such as -L. Libraries (-lfoo) should be added to the LDLIBS variable instead.